src/home_robot/home_robot/perception/detection/grounded_sam/sam_perception.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import os
+import sys
+import time
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import cv2
+import numpy as np
+import pandas as pd
+import supervision as sv
+import torch
+import torchvision
+from groundingdino.util.inference import Model
+
+sys.path.insert(
+    0, str(Path(__file__).resolve().parent / "Grounded-Segment-Anything/EfficientSAM")
```

Comment:
i really dont like this sort of thing, i take it there's no good option though

Reply:
This is just copying the existing code from Grounded SAM. The other option was to implement a new SAM perception in a different folder (and add to install_deps.sh). At some point we might want to find a more permanent solution for including such dependencies -- akin to `submodules` in `accel-cortex`

Reply:
alright, no problem